src/codegen/types/conversions.js

```diff
@@ -188,27 +188,28 @@ const VALUE_TO_ASSEMBLYSCRIPT = [
 const ASSEMBLYSCRIPT_TO_VALUE = [
   // Arrays
 
-  ['Array<Address>', '[Bytes]', code => `Value.fromBytesArray(${code})`],
-  ['Array<Bytes>', '[Bytes]', code => `Value.fromBytesArray(${code})`],
-  ['Array<boolean>', '[Boolean]', code => `Value.fromBooleanArray(${code})`],
-  ['Array<i32>', '[Int]', code => `Value.fromI32Array(${code})`],
-  ['Array<BigInt>', '[BigInt]', code => `Value.fromBigIntArray(${code})`],
-  ['Array<string>', '[String]', code => `Value.fromStringArray(${code})`],
-  ['Array<string>', '[ID]', code => `Value.fromStringArray(${code})`],
-  ['Array<BigDecimal>', '[BigDecimal]', code => `Value.fromBigDecimalArray(${code})`],
-  ['Array<string>', /\[.*\]/, code => `Value.fromStringArray(${code})`],
+  ['Array<Address>', '[Bytes]', code => `Value.fromBytesArray(${code})`, 'new Array(0)'],
+  ['Array<Bytes>', '[Bytes]', code => `Value.fromBytesArray(${code})`, 'new Array(0)'],
+  ['Array<boolean>', '[Boolean]', code => `Value.fromBooleanArray(${code})`, 'new Array(0)'],
+  ['Array<i32>', '[Int]', code => `Value.fromI32Array(${code})`, 'new Array(0)'],
+  ['Array<BigInt>', '[BigInt]', code => `Value.fromBigIntArray(${code})`, 'new Array(0)'],
+  ['Array<string>', '[String]', code => `Value.fromStringArray(${code})`, 'new Array(0)'],
+  ['Array<string>', '[ID]', code => `Value.fromStringArray(${code})`, 'new Array(0)'],
+  ['Array<BigDecimal>', '[BigDecimal]', code => `Value.fromBigDecimalArray(${code})`, 'new Array(0)'],
+  ['Array<string>', /\[.*\]/, code => `Value.fromStringArray(${code})`, 'new Array(0)'],
+  ['Array<string | null>', null, code => `Value.fromStringArray(${code})`, 'new Array(0)'],
 
   // Scalar values
 
-  ['Address', 'Bytes', code => `Value.fromBytes(${code})`],
-  ['Bytes', 'Bytes', code => `Value.fromBytes(${code})`],
-  ['boolean', 'Boolean', code => `Value.fromBoolean(${code})`],
-  ['i32', 'Int', code => `Value.fromI32(${code})`],
-  ['BigInt', 'BigInt', code => `Value.fromBigInt(${code})`],
-  ['string', 'String', code => `Value.fromString(${code})`],
-  ['string', 'ID', code => `Value.fromString(${code})`],
-  ['BigDecimal', 'BigDecimal', code => `Value.fromBigDecimal(${code})`],
-  ['string', /.*/, code => `Value.fromString(${code})`],
+  ['Address', 'Bytes', code => `Value.fromBytes(${code})`, 'Address.zero()'],
+  ['Bytes', 'Bytes', code => `Value.fromBytes(${code})`, 'Bytes.empty()'],
+  ['boolean', 'Boolean', code => `Value.fromBoolean(${code})`, 'false'],
+  ['i32', 'Int', code => `Value.fromI32(${code})`, '0'],
+  ['BigInt', 'BigInt', code => `Value.fromBigInt(${code})`, 'BigInt.zero()'],
+  ['string', 'String', code => `Value.fromString(${code})`, "''"],
+  ['string', 'ID', code => `Value.fromString(${code})`, "''"],
+  ['BigDecimal', 'BigDecimal', code => `Value.fromBigDecimal(${code})`, 'BigDecimal.zero()'],
+  ['string', /.*/, code => `Value.fromString(${code})`, "''"],
```

Comment:
Not sure if this was the best place to put the zero values, I had a separate const for this, but it didn't seem to fit the whole `src/codegen/types/*` architecture. It was something like this:

```js
const DEFAULT_AS_VALUE_CONSTRUCTION = [
  // Arrays

  ['Array<Address>', 'new Array(0)'],
  ['Array<Bytes>', 'new Array(0)'],
  ['Array<boolean>', 'new Array(0)'],
  ['Array<i32>', 'new Array(0)'],
  ['Array<BigInt>', 'new Array(0)'],
  ['Array<string>', 'new Array(0)'],
  ['Array<BigDecimal>', 'new Array(0)'],

  // Scalar values

  ['Address', 'Address.zero()'],
  ['Bytes', 'Bytes.empty()'],
  ['boolean', 'false'],
  ['i32', '0'],
  ['BigInt', 'BigInt.zero()'],
  ['string', "''"],
  ['BigDecimal', 'BigDecimal.zero()'],
]
```

Reply:
We can also discuss if there's any chance we could be missing a type, maybe I should use RegExp for the `Array` type for example.

But I'm a bit confident that it will work in all cases since the function I created (`initializedValueFromAsc`) is almost a copy of `valueFromAsc` which is used for generating the `setters` we have [today](https://github.com/graphprotocol/graph-cli/blob/4959e871bbf4a450b310f0dfbba7c22ce6f739a0/src/codegen/schema.js#L173)